server/server.go

```diff
@@ -148,6 +147,18 @@ func New(opts *Options) *Server {
 	return s
 }
 
+func (s *Server) getOpts() *Options {
+	s.optsMu.RLock()
+	defer s.optsMu.RUnlock()
```

Comment:
This might be expensive for when getOpts is in a fast path. Defer is better, but still slow. Just explicitly unlock like setOpts does.